File: 0205/solution.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        left = {}
        right = {}
        if len(s) != len(t):
            return False

        for i in range(len(s)):
            if s[i] in left:
                if left[s[i]] != t[i]:
                    return False
            else:
                left[s[i]] = t[i]

            if t[i] in right:
                if right[t[i]] != s[i]:
                    return False
            else:
                right[t[i]] = s[i]

        return True

File: 0205/solution.java
import java.util.Map;
import java.util.HashMap;

class Solution {
    public boolean isIsomorphic(String s, String t) {
       if(s.length() != t.length()) return false;
       Map<Character, Character> ss = new HashMap<>();
       Map<Character, Character> tt = new HashMap<>();
       for(int i = 0; i < s.length(); i++) {
            char sChar = s.charAt(i);
            char tChar = t.charAt(i);
            ss.put(sChar, ss.getOrDefault(sChar, tChar));
            tt.put(tChar, tt.getOrDefault(tChar, sChar));
            if(ss.get(sChar) != tChar || tt.get(tChar) != sChar) {
                return false;
            }
       }
       return true;
    }
}